### PR TITLE
feat: Add Diff View component.

### LINKED ISSUE #11: Action Builder - JSON Diff
As a security measure for configurations we store on IPFS and access via ens we need some easy way of verifying json files as someone could very easily sneak in small changes to rich contract json for a malicious attack.
On the UI we will determine if the new and old ipfs files are json when updating ens and then display this component to show a diff of the changes, much like what github does.
It should be very clear and easy to view, maybe in the action dropdown.

### PR DESCRIPTION
**Pull request by mprasanjith**
# Description

Adds a diff view component to the action builder. The current implementation supports showing JSON diffs for the IPFS content hash update actions, but can be tweaked easily to support additional implementations.

A new `useIPFSFileMetadata` hook has also been implemented, which obtains the file size and mime type for an IPFS content-hash without downloading the content. It should be noted that this seems to be mostly undocumented although almost all gateways seem to support it.

Closes #11.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally using unit tests. Storybook was used for testing the UI.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

File: src/hooks/Guilds/ipfs/useIPFSFileMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return { fileSize: Number.parseInt(contentLength), mime: contentType };
+  }
+
+  const response = await Promise.any([
+    fetcher('https://ipfs.io/ipfs/' + hash),
+    fetcher('https://gateway.ipfs.io/ipfs/' + hash),
+    fetcher('https://gateway.pinata.cloud/ipfs/' + hash),
+    fetcher('https://dweb.link/ipfs/' + hash),
+    fetcher('https://infura-ipfs.io/ipfs/' + hash),
+    fetcher('https://cloudflare-ipfs.com/ipfs/' + hash),
```
Comment (Kenny-Gin1) [inline src/hooks/Guilds/ipfs/useIPFSFileMetadata.ts:23]: Cool! Good use of `Promise.any`

**Review by Kenny-Gin1 — COMMENTED** [review]
Nice! I will take a closer look a bit later. But what do you think about providing some screenshots of the end result?

Also, I think you have an integration test failing.

**Comment by rossneilson** [comment]
From my testing
Screenshot 2022-08-04 at 10 25 50

Its a little tough to test right now since we are not able to look up subdomains which is where json data would usually be stored
For now though functionality looks good, just going through code now

**Review by rossneilson — APPROVED** [review]

**Review by Kenny-Gin1 — DISMISSED** [review]

**Review by rossneilson — CHANGES_REQUESTED** [review]
Some merge conflicts and I think pulling develop should fix your integration test

**Review by rossneilson — APPROVED** [review]

**Review by rossneilson — APPROVED** [review]

**Pull request merged**